webdriver/tests/bidi/browsing_context/fragment_navigated/fragment_navigated.py

```diff
@@ -0,0 +1,224 @@
+
+
+async def test_timestamp(bidi_session, current_time, subscribe_events, inline, new_tab, wait_for_event):
+    await subscribe_events(events=[FRAGMENT_NAVIGATED_EVENT])
+
+    time_start = await current_time()
+
+    on_entry = wait_for_event(FRAGMENT_NAVIGATED_EVENT)
+    url = inline("<div>foo</div>")
+    await bidi_session.browsing_context.navigate(context=new_tab["context"], url=url)
+    event = await on_entry
+
+    time_end = await current_time()
+
+    assert_navigation_info(
+        event,
+        {"context": new_tab["context"], "timestamp": int_interval(time_start, time_end)}
+    )
+
+
+async def test_iframe(bidi_session, subscribe_events, new_tab, test_page, test_page_same_origin_frame):
+    events = []
+
+    async def on_event(method, data):
+        # Filter out events for about:blank to avoid browser differences
+        if data["url"] != 'about:blank':
+            events.append(data)
+
+    remove_listener = bidi_session.add_event_listener(
+        FRAGMENT_NAVIGATED_EVENT, on_event
+    )
+    await subscribe_events(events=[FRAGMENT_NAVIGATED_EVENT])
+
+    await bidi_session.browsing_context.navigate(
+        context=new_tab["context"], url=test_page_same_origin_frame
+    )
+
+    wait = AsyncPoll(
+        bidi_session, message="Didn't receive browsingContext.fragmentNavigated for frames"
+    )
+    await wait.until(lambda _: len(events) >= 2)
+    assert len(events) == 2
+
+    contexts = await bidi_session.browsing_context.get_tree(root=new_tab["context"])
+
+    assert len(contexts) == 1
+    root_info = contexts[0]
+    assert len(root_info["children"]) == 1
+    child_info = root_info["children"][0]
+
+    # The ordering of fragmentNavigated events is not guaranteed between the
+    # root page and the iframe, find the appropriate events in the current list.
+    first_is_root = events[0]["context"] == root_info["context"]
+    root_event = events[0] if first_is_root else events[1]
+    child_event = events[1] if first_is_root else events[0]
+
+    assert_navigation_info(
+        root_event,
+        {"context": root_info["context"], "url": test_page_same_origin_frame}
+    )
+    assert_navigation_info(child_event, {"context": child_info["context"], "url": test_page})
+
+    remove_listener()
+
+
+async def test_cross_document(
+    bidi_session, new_tab, url, subscribe_events, wait_for_event
+):
+    await subscribe_events([FRAGMENT_NAVIGATED_EVENT])
+
+    on_frame_navigated = wait_for_event(FRAGMENT_NAVIGATED_EVENT)
+
+    result = await bidi_session.browsing_context.navigate(
+        context=new_tab["context"], url=url(EMPTY_PAGE))
+
+    recursive_compare(
+        {
+            'context': new_tab["context"],
+            'navigation': result["navigation"],
+            'timestamp': any_int,
+            'url': url(EMPTY_PAGE)
+        },
+        await on_frame_navigated,
+    )
+
+
+@pytest.mark.parametrize(
+    "navigation_kind",
+    ["browsing_context.navigate", "script.call_function"],
```

Comment:
Good addition. I think the other navigation related events could also benefit from that. Might be a good follow-up. Maybe you could file at least an issue for wpt?

But I wonder if we should have them as two separate tests without parameterization. Then we could actually parameterize the `test_script` test for `callFunction` and `evaluate`.

Reply:
If you don't want to split this into two distinct tests, can you please at least add id's here so that we have a proper naming of the generated test names? Thanks.

I would still suggest to have two distinct tests given that we do not have a navigation id anyway for the scripting case.

Reply:
I filed https://github.com/web-platform-tests/wpt/issues/40866 To me it seems like have one test makes sense to avoid duplicating the hashes (originally I had two tests for this but decided to merge it)